src/mars.d

```diff
@@ -241,6 +240,31 @@ extern (C++) void genCmain(Scope* sc)
     rootHasMain = sc._module;
 }
 
+/**
+ * Recursevely call semantic3() on import tree nodes.
+ */
+private void semantic3OnDependencies(Module m, bool[string] importArray)
```

Comment:
I replaced the AA with the string-key hashtable in root.springtable. Hope it's ok now. Thanks for the tip